### ISSUE TITLE
`unknown import symbol` when importing submodules of a namespace package with the  submodules in different physical locations.

### ISSUE THREAD

**Issue by tylershunt** [microsoft/pyright#7494]
**Describe the bug**
When a namespace package is spread across a few different directories on the python path, pyright complains if packages from different directories are imported from on the same line. This does not happen when they are imported from on different lines. The python interpreter accepts both, and I think pyright should too.

**Code or Screenshots**
For a minimal repo, set up a structure like this:
```
.
├── dev
│   └── ns_package
│       ├── b
│       │   └── __init__.py
│       └── c
│           ├── __init__.py
│           ├── breaks.py
│           └── works.py
└── lib
    └── ns_package
        └── a
            └── __init__.py
```
All files are empty except for `breaks.py` and `works.py`.

`breaks.py` looks like this:
```python
from ns_package import a, b

_ = (a, b)
```
`works.py` looks like this:
```python
from ns_package import a
from ns_package import b

_ = (a, b)
```
Here's what the output of pyright looks like on each file:
```bash
$ pyright --version
pyright 1.1.354
$ PYTHONPATH=lib:dev pyright ./dev/ns_package/c/works.py
0 errors, 0 warnings, 0 informations
$ PYTHONPATH=lib:dev pyright ./dev/ns_package/c/breaks.py
/Users/thunt/test_repro/dev/ns_package/c/breaks.py
  /Users/thunt/test_repro/dev/ns_package/c/breaks.py:1:27 - error: "b" is unknown import symbol (reportAttributeAccessIssue)
  /Users/thunt/test_repro/dev/ns_package/c/breaks.py:1:27 - error: Type of "b" is unknown (reportUnknownVariableType)
  /Users/thunt/test_repro/dev/ns_package/c/breaks.py:3:1 - error: Type of "_" is partially unknown
    Type of "_" is "tuple[Module("ns_package.a"), Unknown]" (reportUnknownVariableType)
3 errors, 0 warnings, 0 informations
```

**Comment by erictraut** [comment]
Pyright's import system relies on symbols on the same line being imported from different namespace packages. Changing this would require a major overhaul and result in performance degradation in the common case, so it's unlikely that we'd implement this any time soon. Closing as "not planned".
